src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+
+instance FromProto Proto.AccountCreationDetails where
+    type Output' Proto.AccountCreationDetails = ValidResult
+    fromProto acDetails = do
+        ecdAccount <- fromProto =<< acDetails ^? ProtoFields.address
+        ecdRegId <- fmap fst . fromProto =<< acDetails ^? ProtoFields.regId
+        let vrEvents = [AccountCreated ecdAccount, CredentialDeployed{..}]
+        return TxSuccess{..}
+
+instance FromProto Proto.BakerEvent'BakerAdded where
+    type Output' Proto.BakerEvent'BakerAdded = BakerAddedEvent
+    fromProto baEvent = do
+        keysEvent <- fromProto =<< baEvent ^? ProtoFields.keysEvent
+        stake <- fromProto =<< baEvent ^? ProtoFields.stake
+        restakeEarnings <- baEvent ^? ProtoFields.restakeEarnings
+        return (keysEvent, stake, restakeEarnings)
+
+instance FromProto Proto.BakerKeysEvent where
+    type Output' Proto.BakerKeysEvent = BakerKeysEvent
+    fromProto bkEvent = do
+        bakerId <- fromProto =<< bkEvent ^? ProtoFields.bakerId
+        addr <- fromProto =<< bkEvent ^? ProtoFields.account
+        signKey <- fromProto =<< bkEvent ^? ProtoFields.signKey
+        electionKey <- fromProto =<< bkEvent ^? ProtoFields.electionKey
+        aggregationKey <- fromProto =<< bkEvent ^? ProtoFields.aggregationKey
+        return (bakerId, addr, signKey, electionKey, aggregationKey)
+
+instance FromProto Proto.RegisteredData where
+    type Output' Proto.RegisteredData = RegisteredData
+    -- FIXME: Can we just use deMkSerialize here? Types check out, but fromShort
+    -- is used in the other direction...
+    fromProto rData = RegisteredData . BSS.toShort <$> rData ^? ProtoFields.value
+
+instance FromProto Proto.NewRelease where
+    type Output' Proto.NewRelease = (Timestamp, Amount)
+    fromProto nRelease = do
+        tStamp <- fromProto =<< nRelease ^? ProtoFields.timestamp
+        amount <- fromProto =<< nRelease ^? ProtoFields.amount
+        return (tStamp, amount)
+
+instance FromProto Proto.Memo where
+    type Output' Proto.Memo = Memo
+    -- FIXME: Can we just use deMkSerialize here? Types check out, but fromShort
+    -- is used in the other direction...
+    fromProto memo = Memo . BSS.toShort <$> memo ^? ProtoFields.value
+
+instance FromProto Proto.ArInfo'ArIdentity where
+    type Output' Proto.ArInfo'ArIdentity = ArIdentity
+    fromProto arIdentity = do
+        let arId = deMkWord32 arIdentity
+        -- This can not be 0.
+        -- FIXME: This is not checked in the FFI `ar_info_create` function of base.
+        if arId == 0
+            then Nothing
+            else return $ ArIdentity arId
+
+instance FromProto Proto.IpIdentity where
+    type Output' Proto.IpIdentity = IdentityProviderIdentity
+    fromProto arIdentity = do
+        let ipIp = deMkWord32 arIdentity
+        -- FIXME: Can this not be 0 either?
+        -- FIXME: This is not checked in the FFI `ip_info_create` function of base.
+        if ipIp == 0
+            then Nothing
```

Comment:
This can be 0, and is 0 on mainnet and testnet for an identity provider.

There are technical reasons for this being allowed, and not allowed for ArIdentity.